common/src/main/java/com/microsoft/identity/common/internal/ui/webview/challengehandlers/NonceRedirectHandler.kt

```diff
@@ -0,0 +1,70 @@
+package com.microsoft.identity.common.internal.ui.webview.challengehandlers
+
+import android.app.Activity
+import androidx.annotation.VisibleForTesting
+import com.microsoft.identity.common.CommonPrtCredentialHolder
+import com.microsoft.identity.common.adal.internal.util.StringExtensions
+import com.microsoft.identity.common.java.util.StringUtil
+import com.microsoft.identity.common.logging.Logger
+import java.net.MalformedURLException
+import java.net.URL
+
+/**
+ * Handler for processing nonce from redirect and attaching new prt credential header on web view.
+ */
+class NonceRedirectHandler {
```

Comment:
>NonceRedirectHandler [](http://example.com/codeflow?start=6&length=20)

NonceRedirectHandler(private val rtcProvider : IRefreshTokenCredentialProvider):  IChallengeHandler<URL, HashMap<>>

URL -> redirect URI containing nonce etc
String -> (response) header map updated with new refresh token credential

OR
NonceRedirectHandler(private val rtcProvider : IRefreshTokenCredentialProvider, private val webview: Webview):  IChallengeHandler<URL, HashMap<>>

in processChallenge, continue using webview object (similar to PKeyAuthChallengeHandler)